static/js/disaster_dashboard/map_section.tsx

```diff
@@ -0,0 +1,252 @@
+      props.geoJson.properties.current_geo !==
+        props.selectedPlaceInfo.place.dcid
+    ) {
+      return;
+    }
+    draw();
+    removeSpinner(CONTENT_SPINNER_ID);
+  }, [props.disasterEventPoints, props.geoJson]);
+
+  if (_.isEmpty(props.geoJson)) {
+    removeSpinner(CONTENT_SPINNER_ID);
+    return (
+      <div className="disaster-dashboard-map-section">
+        <div className="error-message">
+          Sorry, we do not have maps for this place.
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className="disaster-dashboard-map-section">
+      <div className="zoom-button-section">
+        <div id={ZOOM_IN_BUTTON_ID} className="zoom-button">
+          <i className="material-icons">add</i>
+        </div>
+        <div id={ZOOM_OUT_BUTTON_ID} className="zoom-button">
+          <i className="material-icons">remove</i>
+        </div>
+      </div>
+      <div id={MAP_ID} ref={svgContainer}></div>
+      {props.selectedDisaster === DisasterType.ALL ? (
+        <div className="all-disasters-legend">
+          {Object.keys(DISASTER_COLORS).map((disasterType) => {
+            return (
+              <div className="all-disasters-legend-entry" key={disasterType}>
+                <div
+                  className="all-disasters-legend-color"
+                  style={{ backgroundColor: DISASTER_COLORS[disasterType] }}
+                ></div>
+                <span>{disasterType}</span>
+              </div>
+            );
+          })}
+        </div>
+      ) : (
+        <div id={LEGEND_CONTAINER_ID}></div>
+      )}
+      <div id="disaster-dashboard-info-card" ref={infoCardRef} />
+    </div>
+  );
+
+  function onPointClicked(
+    point: DisasterEventPoint,
+    event: PointerEvent
+  ): void {
+    const infoCard = d3.select(infoCardRef.current);
+    ReactDOM.render(
+      <InfoCard
+        onClose={() =>
+          d3.select(infoCardRef.current).style("visibility", "hidden")
+        }
+        eventData={point}
+      />,
+      infoCardRef.current
+    );
+    const infoCardHeight = (
+      infoCard.node() as HTMLDivElement
+    ).getBoundingClientRect().height;
+    const infoCardWidth = (
+      infoCard.node() as HTMLDivElement
+    ).getBoundingClientRect().width;
+    const containerWidth = (
+      d3.select(svgContainer.current).node() as HTMLDivElement
+    ).getBoundingClientRect().width;
+    let left = Math.min(event.offsetX, containerWidth - infoCardWidth);
+    if (left < 0) {
+      left = 0;
+      infoCard.style("width", containerWidth + "px");
+    } else {
+      infoCard.style("width", "fit-content");
+    }
+    let top = event.offsetY - infoCardHeight - iNFO_CARD_OFFSET;
+    if (top < 0) {
+      top = event.offsetY + iNFO_CARD_OFFSET;
+    }
+    infoCard
+      .style("left", left + "px")
+      .style("top", top + "px")
+      .style("visibility", "visible");
+  }
+
+  function canClickRegion(placeDcid: string): boolean {
+    const enclosingPlace =
```

Comment:
please add a comment on why europe is special-cased

Reply:
done